### DISCUSSION TITLE
Raspberry Pi OS Now Shipping With Vulkan Support By Default

### DISCUSSION THREAD

**Discussion by alatteri** [CESNET/UltraGrid#381] [General]
This might be helpful to increase performance of UG on Pi?

https://www.phoronix.com/news/Raspberry-Pi-OS-Default-V3DV

**Comment by MartinPulec** [comment]
well, I know that the question is over the year old, but since no-one reacted...

I am unfortunately not familiar with Mesa Vulkan APIs, especially for RPi. UltraGrid ARM AppImages should build with Vulkan (display) support. I don't know whether that information has some implication to UltraGrid from the development perspective, I believe that this is rather the case for the Vulkan runtime so it this should work out-of-box.

Without deeper knowledge yes, I suppose that if rendering performance is poor with other SW displays, using Vulkan may help. (AFAIK even OpenGL on RPI was just emulated, there was only OpenGL ES natively; SDL with EGL backend may perform).

I haven't tested this yet because I don't unfortunately have an OS appropriate image flashed by now but I can try to if needed.
